opensearch/sycamore-opensearch.sh

```diff
@@ -226,6 +230,78 @@ END
     echo "EMBEDDING_MODEL_ID=${EMBEDDING_MODEL_ID}" >>"${PERSISTENT_ENV_TMP}"
 }
 
+sp_setup_reranking_model() {
+    local all_config=$(jq '@json' <<END
+{
+  "_name_or_path": "BAAI/bge-reranker-base",
+  "architectures": [
+    "XLMRobertaForSequenceClassification"
+  ],
+  "attention_probs_dropout_prob": 0.1,
+  "bos_token_id": 0,
+  "classifier_dropout": null,
+  "eos_token_id": 2,
+  "hidden_act": "gelu",
+  "hidden_dropout_prob": 0.1,
+  "hidden_size": 768,
+  "id2label": {
+    "0": "LABEL_0"
+  },
+  "initializer_range": 0.02,
+  "intermediate_size": 3072,
+  "label2id": {"LABEL_0": 0},
+  "layer_norm_eps": 1e-05,
+  "max_position_embeddings": 514,
+  "model_type": "xlm-roberta",
+  "num_attention_heads": 12,
+  "num_hidden_layers": 12,
+  "output_past": true,
+  "pad_token_id": 1,
+  "position_embedding_type": "absolute",
+  "torch_dtype": "float32",
+  "transformers_version": "4.33.3",
+  "type_vocab_size": 1,
+  "use_cache": true,
+  "vocab_size": 250002
+}
+END
+)
+
+    local file="${ARYN_STATUSDIR}/curl.reranking_model"
+    _curl_json -XPOST "${BASE_URL}/_plugins/_ml/models/_register" \
+        -o "${file}" \
+        --data @- <<END || die "Error registering reranker"
+{
+  "name": "BAAI/bge-reranker-base-quantized",
+  "version": "1.0.0",
+  "description": "Cross Encoder text similarity model",
+  "model_format": "ONNX",
+  "function_name": "TEXT_SIMILARITY",
+  "model_group_id": "${MODEL_GROUP_ID}",
+  "model_content_hash_value": "04157d66d847d08b3d2b51ad36cf0e1fb82afadb8086212a1d2bac2b7d6fe08a",
+  "model_config": {
+    "model_type": "roberta",
+    "embedding_dimension": 1,
+    "framework_type": "huggingface_transformers",
+    "all_config": ${all_config}
+  },
+  "url": "https://aryn-public.s3.amazonaws.com/models/BAAI/bge-reranker-base-quantized-2.zip"
```

Comment:
Are we not using the one hosted by OpenSearch (assuming they will host it)?

Reply:
They're just hosting MiniLM-L-6-v2 and MiniLM-L-12-v2. I tried to get Dhrubo to take this one but he didn't want to. This one should be better tho. But we should run 'em all through our benchmarks